pages/lua_manual/api_reference.md

```diff
@@ -0,0 +1,1849 @@
+---
+layout: page
+title: Lua API Reference
+status: improve
+---
+
+All instantiable classes define a `new()` method which returns an object of that particular class. 
+
+With the exception of this `new()` method, all members listed will be member methods.
+
+## ElementDataGrid
+
+Inherits: `Element`{: .type }
+
+ElementDataGrid derives from Element. The data grid has the following methods and properties:
+
+### Properties
+
+| Types | Name |
+| ------------ | ---- |
+| `ElementDataGridRow`{: .type } | rows |
+
+
+### Methods
+
+| Return value | Name |
+| ------------ | ---- |
+| `nil`{: .type } | AddColumn(`string`{: .type } fields, `string`{: .type } formatter, `number`{: .type } initial_width, `string`{: .type } header_rml) |
+| `nil`{: .type } | SetDataSource(`string`{: .type } data_source_name) |
+
+
+### Metafunctions
+
+
+
+
+### Property Descriptions
+
+*  `ElementDataGridRow`{: .type } rows
```

Comment:
```suggestion
* <span id="ElementDataGrid-AddColumn" class="lua_method"><span class="lua_type">ElementDataGridRow</span> rows</span>
```
For linking to it, and styling it.

Reply:
I ended up with getting the same result with this: 
```md
####  `ElementDataGridRow`{: .lua-type } <a name='ElementDataGrid-rows'>rows</a>{: .lua-method }
```

Reply:
Is this `<a name>` standard html? I'm not sure why we don't just use a span and id?

Reply:
The name attribute in anchor tag also allows us to give the text link formatting. Although I am having second thoughts on this. I ended up giving both name and href attributes to these, where I think name attribute becomes redundant. 
I was initially wanting to give the option to just right click > copy link and share it elsewhere, whether it is a property or a function. Maybe just href would be enough?
I could just do it like you mentioned, with a span and an id but that doesn't allow the user to copy the link directly. They will have to scroll up to the table and then copy the link from there. This can be avoided with we made it a link right there.

Reply:
I see, copying like that would indeed be useful. I'll leave it to you to figure out :)

Reply:
Well I had to add both `name` and `href` to make it linkable AND to be able to copy the link from it. This should work fine